WordPress/Classes/Extensions/AbstractPost+Dates.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+extension AbstractPost {
+
+    /// The string to display to the user representing a post's date.
+    ///
+    /// - **Scheduled**: Displays time + date
+    /// - **Immediately**: Displays "Publish Immediately" string
+    /// - **Published or Draft**: Shows relative date when < 7 days
+    public func displayDate() -> String? {
+        let context = managedObjectContext ?? ContextManager.sharedInstance().mainContext
+        let blogService = BlogService(managedObjectContext: context)
+        let timeZone = blogService.timeZone(for: blog)
+
+        if originalIsDraft() || status == .pending {
+            return dateModified?.mediumString(timeZone: timeZone)
+        } else if isScheduled() {
```

Comment:
Since each `if` has a `return`, the `else`s aren't really necessary.

Reply:
I often leave the `else` in to make it clear that there isn't one scope that isn't actually returning (easier to skim). I went ahead and removed and added some comments explaining the intent of each block.